tests/pyfunc/test_pyfunc_schema_enforcement.py

```diff
@@ -2077,6 +2077,44 @@ def predict(self, context, model_input, params=None):
     np.testing.assert_equal(result, expected_df.values.tolist()[0])
 
 
+def test_input_example_validation_during_logging(tmp_path):
```

Comment:
Regarding to [this comment](https://github.com/mlflow/mlflow/pull/12625#discussion_r1675384699), while it's technically hard to sync OSS<->Databricks server, we can probably add some guardrail so that at least the conversion logic is consistent with OSS server.
 
1. Add different input data for this test case.
2. [Follow-up PR] In each flavor there are a few test cases that runs `pyfunc_serve_and_score_model`. Update them to use saved example instead of hand-crafted payload.

Can we do that?